digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala

```diff
@@ -402,80 +404,45 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
-      val defaultTolerance = 2
-      ((urbanAreaAddressMValue - addressSectionAddressMValue).abs) > defaultTolerance
-    }
-
-    def createSectionPart(roadAddress: ViiteRoadAddress, section: AddressSection, assetType: TRTrafficSignType, assetValue: String) = {
-      calculateMeasures(roadAddress, section) match {
-        case Some(measure) =>
-          val newTrAsset = TierekisteriTrafficSignData(section.roadNumber, section.roadPartNumber, section.roadPartNumber, section.track, section.startAddressMValue, section.startAddressMValue, roadSide, assetType, assetValue)
-          createLinearAsset(roadLink, roadAddress, section, measure, newTrAsset)
-        case _ => None
+    def generateUrbanTrafficSign(trAsset: TierekisteriUrbanAreaData): TierekisteriTrafficSignData= {
+      if (trAsset.assetValue == notUrbanArea) {
+        TierekisteriTrafficSignData(trAsset.roadNumber, trAsset.startRoadPartNumber, trAsset.endRoadPartNumber, trAsset.track, trAsset.startAddressMValue, trAsset.startAddressMValue, roadSide, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString)
+      } else {
+        TierekisteriTrafficSignData(trAsset.roadNumber, trAsset.startRoadPartNumber, trAsset.endRoadPartNumber, trAsset.track, trAsset.startAddressMValue, trAsset.startAddressMValue, roadSide, TRTrafficSignType.UrbanArea, urbanAreaSpeedLimit.toString)
       }
     }
 
-    //Initial Part of The Section Splited
-    if ((trUAAsset.startAddressMValue > addressSection.startAddressMValue) && (verifyDefaultTolerance(trUAAsset.startAddressMValue, addressSection.startAddressMValue)) && (trUAAsset.startAddressMValue <= addressSection.endAddressMValue.head)) {
-      val newStartAddressSection = addressSection.copy(endAddressMValue = Some(trUAAsset.startAddressMValue))
-      createSectionPart(roadAddress, newStartAddressSection, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString)
+    def getDefaultTrTrafficSign(addressSection: AddressSection): Option[TierekisteriAssetData] ={
+      Some(TierekisteriTrafficSignData(addressSection.roadNumber, addressSection.roadPartNumber, addressSection.roadPartNumber, addressSection.track, addressSection.startAddressMValue, addressSection.startAddressMValue, roadSide, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString))
     }
 
-    //Middle Part of The Section Splited
-    if ((trUAAsset.startAddressMValue >= addressSection.startAddressMValue) && (trUAAsset.endAddressMValue <= addressSection.endAddressMValue.head)) {
-      val newMiddleAddressSection = addressSection.copy(startAddressMValue = trUAAsset.startAddressMValue, endAddressMValue = Some(trUAAsset.endAddressMValue))
-      createSectionPart(roadAddress, newMiddleAddressSection, TRTrafficSignType.UrbanArea, urbanAreaSpeedLimit.toString)
-    }
+    val trUrbanAssets = trUrbanAreaAssets.filter(_.startRoadPartNumber == addressSection.roadPartNumber)
```

Comment:
Here we are just take care of the start road part number, the urban area is a linear asset like, so we should taking care of the start and the end. The idea of reusing the existing code for the traffic signs is good we just need to convert in the urban area linear asset in two point urban area, start and the end.